via/configuration.py

```diff
@@ -0,0 +1,102 @@
+"""Tools for reading in configuration."""
+
+from inflection import camelize
+
+# pylint: disable=too-few-public-methods
+
+
+class Configuration:
+    """Extracts configuration from params."""
+
+    # Certain configuration options include URLs which we will read, write or
+    # direct the user to. This would allow an attacker to craft a URL which
+    # could do that to a user, so we whitelist harmless parameters instead
+    # From: https://h.readthedocs.io/projects/client/en/latest/publishers/config/#config-settings
+    H_CONFIG_WHITELIST = {
+        # Things we use now
+        "openSidebar",
+        "requestConfigFromFrame",
+        # Things which seem safe
+        "enableExperimentalNewNoteButton",
+        "externalContainerSelector",
+        "focus",
+        "showHighlights",
+        "theme",
+    }
+
+    @classmethod
+    def extract_from_params(cls, params):
+        """Extract Via and H config from query parameters.
+
+        :param params: A mapping of query parameters
+        :return: A tuple of Via, and H config
+        """
+
+        via_params = cls._unflatten(params)
+        h_params = via_params.pop("h", {})
+
+        cls._to_camel_case(h_params)
+        cls._filter_h_params(h_params)
+        cls._move_legacy_params(via_params, h_params)
+
+        # Set some defaults
+        h_params["appType"] = "via"
+        h_params.setdefault("showHighlights", True)
+
+        return via_params, h_params
+
+    @staticmethod
+    def _unflatten(params):
```

Comment:
It's specialised, but it's also private to this class, so it only needs to differentiate itself amongst other functions here, rather than globally.

This function actually returns both `h` and `via` configuration, so I don't think it's accurate to put "Via" in the name. In general, as we _are_ via, it can be broadly assumed that without stating we're in a "Via" context for most things. So this object is "Configuration" rather than "ViaConfiguration" as it's obvious from context.

Reply:
I've updated the doc string to give details of the 'via' key skipping behavior.